frontend/talentsearch/resources/js/components/communityExperienceForm/CommunityExperienceForm.tsx

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { useWatch } from "react-hook-form";
+import { Input } from "@common/components/form/Input";
+import { Checkbox } from "@common/components/form/Checkbox";
+import { errorMessages } from "@common/messages";
+
+export const CommunityExperienceForm: React.FunctionComponent = () => {
+  const intl = useIntl();
+
+  // to toggle whether End Date is required, the state of the Current Role checkbox must be monitored and have to adjust the form accordingly
+  const isCurrent = useWatch({ name: "current-role", defaultValue: false });
+  // ensuring end date isn't before the start date, using this as a minimum value
+  const startDate = useWatch({ name: "start-date" });
```

Comment:
The input name should match the CommunityExperience type name for when we submit the data. So, this should be changed to `startDate`. Also, same goes for `endDate`.